### ISSUE TITLE
RenderQuotedNames.ALWAYS does not work correctly for stored functions

### ISSUE THREAD

**Issue by jonnybbb** [jOOQ/jOOQ#11055]
### Configuration
```
var settings = new Settings();
settings.setRenderQuotedNames(RenderQuotedNames.ALWAYS);
```

### Expected behavior
Routines should properly be quoted with `"` 

```
@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@  Thank you for using jOOQ 3.13.6
                                      
22:40:01,899 DEBUG [org.jooq.tools.LoggerListener                     ] - Executing query          : select "foo" from "mcve"."foo"("i" := ?)
22:40:01,900 DEBUG [org.jooq.tools.LoggerListener                     ] - -> with bind values      : select "foo" from "mcve"."foo"("i" := 1)
```

### Actual behavior
Routines are triple quoted with `"""`

```
@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@  Thank you for using jOOQ 3.14.3

22:41:58,395 DEBUG [org.jooq.tools.LoggerListener                     ] - Executing query          : select "foo" from """mcve"".""foo"""("i" := ?)
22:41:58,396 DEBUG [org.jooq.tools.LoggerListener                     ] - -> with bind values      : select "foo" from """mcve"".""foo"""("i" := 1)
22:41:58,411 DEBUG [org.jooq.tools.LoggerListener                     ] - Exception                
org.jooq.exception.DataAccessException: SQL [select "foo" from """mcve"".""foo"""("i" := ?)]; ERROR: function "mcve"."foo"(i => integer) does not exist
  Hint: No function matches the given name and argument types. You might need to add explicit type casts.
```

### Steps to reproduce the problem

- An MCVE can be helpful to provide a complete reproduction case: https://github.com/jonnybbb/jOOQ-mcve

### Versions

- jOOQ: 3.14.3
- Java: 11
- Database (include vendor): Postgresql 12.5
- OS: Mac OSx, Linux
- JDBC Driver (include name if inofficial driver): org.postgresql:postgresql:42.2.18

**Comment by lukaseder** [comment]
(A late) thank you very much for your report. I can reproduce this and will fix this right away.

**Comment by lukaseder** [comment]
There seem to be some leftovers from a previous refactoring: https://github.com/jOOQ/jOOQ/issues/5997

**Comment by lukaseder** [comment]
Fixed for jOOQ 3.15.0 and 3.14.10 (#11792)
